### DISCUSSION TITLE
Trouble with using newer FF as opposed to older versions

### DISCUSSION THREAD

**Discussion by Mitsurugi-w** [keirf/flashfloppy#659] [General]
Hello.

We have been using FF for years for our Sega System 24 Multigame to replace the floppy drives that failed in arcades years ago.
We had been using the very old v.0.9.19a because it just worked.  
Well due to the chips changing on the Goteks we have obviously been forced to use the newer versions. 
I received a batch of Goteks with the SFRKC30.AT2 pcb.  I had no problem flashing FF 3.30 to it.  
I also tried the latest Development version.  Both are giving me the same issue.

While using the old FF version we had these files on the USB stick:

DSKA0000.hfe
DSKA0001.hfe
DSKA0002.hfe
DSKA0003.hfe
DSKA0004.hfe
DSKA0005.hfe
DSKA0006.hfe
DSKA0007.hfe
DSKA0008.hfe
DSKA0009.hfe
HXCSDFE.cfg

When using a stick with these files I got E31.  I saw that this means I needed FF.cfg so I copied that to the root as well.  It still didn't work so I removed the HXCSDFE,cfg but left FF.CFG and finally games began to load.  However it will ONLY load games 0-3.  If I select any game after 3 it errors.  I've tried new faster USB sticks to no avail. Reformatting, etc.  I tried changing Navigation in the FF.CFG to every option with no change.

Anyone have a clue why this is being a pill?

**Comment by keirf** [comment]
You can't copy HXCSDFE.CFG between USB sticks if it contains slot assignments. If it is simply configuring indexed mode you can do that with FF.CFG instead and that's preferred.

**Comment by keirf** [comment]
Also what error do you get if you select "any game after 3"? You may want to zip up your usb drive and upload contents for me to look at. Either here or you can email me a link.

**Comment by Mitsurugi-w** [comment]
Ok.  So more testing has found that it's not every game after 3 that fails to load.  Only games 4 and 5 fail to load.  These worked fine on the older FF with the STM chip.  What we have determined is that these two have a different track size than the other games.  But like I said before they worked on the older setup.

**Comment by keirf** [comment]
How much different track size? Is there a standard track layout or do the track layouts differ?

**Comment by keirf** [comment]
Does an STM Gotek running latest firmware (3.32) work OK?

**Comment by Mitsurugi-w** [comment]
The working track size are 0x2d00.

The non-working are 0x2f00.

I unfortunately cant test the new FF on an STM Gotek because I sent my personal drive away a few weeks ago so the customer wouldn't need to wait for the new batch to arrive.  I had no idea the STM version had become unobtainium.  :(
[s24multiusb.zip](https://github.com/keirf/flashfloppy/files/8938564/s24multiusb.zip)

**Reply by keirf** [reply]
How do the non-working ones fail? I assume there's no Gotek error, it manifests as unreadable / bad disk on the host that the Gotek is connected to?

**Reply by Mitsurugi-w** [reply]
That is correct, sir.  I should have mentioned that earlier.  Sorry.

**Comment by keirf** [comment]
Okay the problem is that the high-density HFE is marginal on the new Goteks due to reduced RAM for buffering. The new Goteks have only 32kB RAM compared with 64kB on the STM. There are a couple of things we can do:
1. I have remade your longer HFEs (these are 1.88M Sega images) with shorter track length. The data still fits. Perhaps this will get it working, just about.
2. As luck would have it, a new Gotek model using the AT32F435 chip is currently in mass production, and will be ready for sale within the next week. This chip has 384kB RAM! It is much more suited for high-density HFE. Perhaps however this is annoying if you have bought a lot of SFRKC30.AT2!

Anyway, please find the non-working images 4 and 5 remade in the following zip file:
[modified.zip](https://github.com/keirf/flashfloppy/files/8938640/modified.zip)

**Reply by keirf** [reply]
My honest advice however is to sell your AT2 Goteks or PCBs, and hold out for a batch with the new 435 chip. Or at least buy a 435-based Gotek for evaluation.

**Comment by Mitsurugi-w** [comment]
Unfortunately the modified files do not work either.  Thanks for the effort though.

I was hoping to still use these five as I've already done an electrical mod needed to work on the S24 arcade pcb.  This mod doesn't affect the operation of the Gotek otherwise.  I guess I can reverse the mod and try to sell them.  Just hate making the customers wait even longer.

**Reply by keirf** [reply]
It may be due to the fact that the 1.88M format has an 8kB sector, and the longer HFE track length is simply a red herring. Especially if the host is trying to write an 8kB sector, that's basically not going to work with FlashFloppy on the Artery 415 chip. You may have read-only luck with a different USB drive. But like I say, it will always be marginal because I have no more plans to improve 32kB support now that the 435-based Goteks are so close.

If you are buying a batch of 10+ then I think you can order direct from factory.

**Reply by keirf** [reply]
The factory contact is Yan, email guiie@163.com

Tell him I sent you, asking for 435-based Goteks. He'll let you know price, moq, and when exactly they'll be ready to ship.

**Reply by keirf** [reply]
He will even flash them for you. I need to do a FF release that supports 435. It is ready to go, just not officially released yet.

**Comment by Mitsurugi-w** [comment]
How do you order direct?

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by keirf** [reply]
See above, my reply overlapped your question
